src/QLTypes.h

```diff
@@ -80,11 +80,19 @@ typedef struct LongDouble final {
 #else
 typedef long double LongDouble;
 #endif
+// #17: Added decode_bytes and decode_key_part to update explain output with user readable keys
+
+/* In case query has '$gt' or '$lt' operator then field elements are represent in range,
+ * like '//x01e < x < 1' (for '$lt') and '1 < x < //x01f' (for '$gt'), where //x01e represent.
+ * To determine those bytes, enable a check for '0x1f' in 'DVTypeCode' as SPL_CHAR,
+ * for '0x1e' in 'DVTypeCode' already enabled as 'NUMBER'.
+ */
 
 enum class DVTypeCode : uint8_t {
 	MIN_KEY = 0,
 	NULL_ELEMENT = 20,
 	NUMBER = 30,
+	SPL_CHAR = 31, // ASCII value for SPL_CHAR = 31 is '\0x1f'
```

Comment:
Predicates like `$gt`, `$gte`, `$lt`, `$lte` etc.. use `RANGE` type predicate. To create a unended bound we create element with `type code` + 1, to match everything in that range or vice versa. Range can be on any type, not necessary just on numbers. For example, this code breaks on a query like

```
db.coll.find({'a': {'$gt': 'hello'}}).explain()
```

Which is a valid query. You can look at the following code to get better understanding of how ranges are created with one bound.

https://github.com/FoundationDB/fdb-document-layer/blob/master/src/ExtOperator.actor.cpp#L72

Reply:
@apkar We think RANGE would have some limitations for end key calculations. In Explain, we will not be able to give detailed output for inputs like $gt 10, $gt abc etc... because RANGE adds +1 to the typecode and returns as string. Whereas, if we add typecode + 1 for each of the enum in DVTypeCode, then we can handle end limit for any type and also the decode_bytes would work as expected. This will be similar to OBJECT-PACKED_OBJECT and ARRAY-PACKED_ARRAY. What do you think?